Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift

```diff
@@ -139,9 +139,7 @@ public final class AutogeneratedSchemesProjectMapper: ProjectMapping {
 
     private func isAssociatedTestTarget(for target: Target, potentialTestTarget: Target) -> Bool {
         let isTestsBundle = potentialTestTarget.product.testsBundle
-        let isNameMatching = potentialTestTarget.name == "\(target.name)Tests" ||
-            potentialTestTarget.name == "\(target.name)UITests" ||
-            potentialTestTarget.name == "\(target.name)IntegrationTests"
+        let isNameMatching = potentialTestTarget.name.range(of: #"\#(target.name)\w*Tests"#, options: .regularExpression) != nil
```

Comment:
I am far from a regex expert, so please suggest improvements if you have any!

Reply:
That looks correct to me.

Reply:
Perhaps a regex isn't needed here - a slightly faster alternative that achieves the same result is

```swift
potentialTestTarget.name.hasPrefix(target.name) && potentialTestTarget.name.hasSuffix("Tests")
```

Reply:
I like that. It's more immediately clear. I'll update and fix formatting.